powerscale/helper/network_setting_helper.go

```diff
@@ -36,21 +38,70 @@ func UpdateNetworkSettingDataSourceState(ctx context.Context, settingState *mode
 	}
 
 	settingState.SBREnabled = types.BoolValue(settingResponse.Sbr)
-	settingState.ID = types.StringValue("network_setting_datasource")
+	settingState.ID = types.StringValue("network_settings_datasource")
 
 	return
 }
 
-// GetNetworkSetting returns network setting detail.
+// UpdateNetworkSettingResourceState updates resource state.
+func UpdateNetworkSettingResourceState(ctx context.Context, settingState *models.NetworkSettingModel, settingResponse *powerscale.V12NetworkExternalSettings) (err error) {
```

Comment:
Will function return error?

Reply:
@baoy1  good catch,
The CopyFields function was used here. Will remove this err

Reply:
@baoy1  updated